temporallib/auth/auth.py

```diff
@@ -13,77 +13,50 @@
 from google.oauth2 import service_account
 import google.auth.transport.requests
 from typing import Union
-from dataclasses import asdict
-import os
+from pydantic_settings import BaseSettings
+from pydantic import Field, BaseModel
 
 
-class MacaroonAuthOptions:
-    """
-    Defines the parameters for authenticating with Candid.
-    """
+class MacaroonAuthOptions(BaseSettings):
+    macaroon_url: str
+    username: str
+    keys: Optional[KeyPair]
 
-    macaroon_url: str = None
-    username: str = None
-    keys: KeyPair = None
+    class Config:
+        env_prefix = 'TEMPORAL_CANDID_'
 
-    def __init__(self, macaroon_url: str = None, username: str = None, keys: KeyPair = None):
-        self.macaroon_url = macaroon_url or os.getenv("TEMPORAL_CANDID_URL")
-        self.username = username or os.getenv("TEMPORAL_CANDID_USERNAME")
-        self.keys = keys or KeyPair()
 
+class GoogleAuthOptions(BaseSettings):
+    type: str = "service_account"
```

Comment:
question: Did you test this? I thought fields with default values need to come last in Pydantic.

Reply:
I've tested this with both approaches (passing parameters and setting environment variables), and i've also confirmed this through the unit tests.